iceoryx2/src/port/unable_to_deliver_strategy.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2025 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use serde::{de::Visitor, Deserialize, Serialize};
+
+/// Defines the strategy a sender shall pursue when the buffer of the receiver is full
+/// and the service does not overflow.
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+pub enum UnableToDeliverStrategy {
+    /// Blocks until the [`crate::port::subscriber::Subscriber`] has consumed the
+    /// [`crate::sample::Sample`] from the buffer and there is space again
+    Block,
+    /// Do not deliver the [`crate::sample::Sample`].
+    DiscardSample,
+}
+
+impl Serialize for UnableToDeliverStrategy {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&std::format!("{:?}", self))
+    }
+}
```

Comment:
Is this really required? I can't recall that I had to do this the last time I used serde

Reply:
I think so. I did not change the code, just moved it into the new file.